python/pyspark/sql/pandas/conversion.py

```diff
@@ -241,21 +241,81 @@ def toPandas(self) -> "PandasDataFrameLike":
         else:
             return pdf
 
-    def _collect_as_arrow(self, split_batches: bool = False) -> List["pa.RecordBatch"]:
```

Comment:
This isn't an API; I wouldn't fix it for external purposes.

Reply:
Ok, understood. I will make some changes here to reflect that.

P.S. The git diff here is confusing and makes it look like I am proposing a rename and full reimplementation of `_collect_as_arrow` which I am not :)